### PR TITLE
Add source string via the Python API

### PR DESCRIPTION
**Pull request by superDross**
- Source stings can now be added if the component is a monolingual component
- Altered preparation of test file names to work with multiple kwargs

### CODE REVIEW

File: wlc/__init__.py [hunk 1]
```diff
@@ -443,6 +443,10 @@ def changes(self):
     def delete(self):
         self.weblate.raw_request("delete", self._url)
 
+    def add_source_string(self, component, msgid, msgstr):
+        """Adds a source string to a monolingual base file"""
+        return self.weblate.add_source_string(self.slug, component, msgid, msgstr)
+
```
Comment (iBurnette) [inline wlc/__init__.py]: This confuses me a bit. Why add this here and not directly on the component?

File: wlc/__init__.py [hunk 2]
```diff
@@ -195,6 +206,22 @@ def list_languages(self):
         """List languages in the instance."""
         return self.list_factory("languages/", Language)
 
+    def _is_component_monolingual(self, path):
+        comp = self.get_component(path)
```
Comment (iBurnette) [inline wlc/__init__.py]: Missing docstring

File: wlc/__init__.py [hunk 3]
```diff
@@ -131,6 +138,10 @@ def post(self, path, **kwargs):
         """Perform POST request on the API."""
         return self.request("post", path, kwargs)
 
+    def _post_factory(self, prefix, path, kwargs):
+        resp = self.post("/".join((prefix, path, "")), **kwargs)
```
Comment (iBurnette) [inline wlc/__init__.py]: Missing docstring

File: wlc/test_data/api/translations-hello-android-en-units--POST--key=test-monolingual--value=test-me [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+ "language": {"aliases": ["english", "en_en", "base", "source", "eng"],
+              "code": "en",
+              "direction": "ltr",
+              "name": "English",
+              "plural": {"formula": "n != 1", "id": 82, "number": 2, "source": 0, "type": 1},
+              "statistics_url": "http://127.0.0.1:8000/api/languages/en/statistics/",
+              "url": "http://127.0.0.1:8000/api/languages/en/",
+              "web_url": "http://127.0.0.1:8000/languages/en/"},
+ "language_code": "en",
+ "last_author": "Approver Man",
```
Comment (iBurnette) [inline wlc/test_data/api/translations-hello-android-en-units--POST--key=test-monolingual--value=test-me:69]: A+ name

File: wlc/__init__.py (hunk 1 above)
Comment (superDross) [reply wlc/__init__.py]: Good catch! Have implemented as suggested.

File: wlc/__init__.py (hunk 3 above)
Comment (superDross) [reply wlc/__init__.py]: Added the docstring.

File: wlc/__init__.py (hunk 2 above)
Comment (superDross) [reply wlc/__init__.py]: Added the docstring.

File: wlc/__init__.py [hunk 5]
```diff
@@ -53,6 +53,13 @@ def __init__(self):
         super().__init__("Access denied, API key is wrong or missing")
 
 
+class IsNotMonolingual(WeblateException):
```
Comment (kamilgrymuza) [inline wlc/__init__.py:56]: Very good idea to catch this and raise an exception 👍

**Comment by kamilgrymuza** [comment]
Looks good @superDross 👍

**Pull request merged**